Week4/app/src/main/java/com/oliverperez/blockbusterchallenge/Adapters/MovieGridAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.oliverperez.blockbusterchallenge.Adapters
+
+import android.media.Image
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.RecyclerView
+import com.oliverperez.blockbusterchallenge.DataManagers.MovieDataManager
+import com.oliverperez.blockbusterchallenge.Models.Movie
+import com.oliverperez.blockbusterchallenge.R
+import com.oliverperez.blockbusterchallenge.Views.MovieViewHolder
+import kotlinx.android.synthetic.main.movie_view_holder.view.*
+
+class MovieGridAdapter(private val movies: List<Movie>, private val clickListener: MovieClickListener): RecyclerView.Adapter<MovieViewHolder>() {
+    interface MovieClickListener {
```

Comment:
Creating interface for click listeners is okay, and that's what the course taught you. But you can also do the same with lambdas. I will show you on Thursday how to do that.